### PR TITLE
Interface tests now run on Windows, Mac and Linux.

### PR DESCRIPTION
**Pull request by serban-nicusor-toptal**
#### Submisison Checklist

- [ ] Run tests: `./runCmdStanTests.py src/test`
- [ ] Declare copyright holder and open-source license: see below

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -3,17 +3,23 @@ import org.stan.Utils
...
-def setupCXX(CXX = env.CXX) {
+def setupCXX(failOnError = true, CXX = env.CXX) {
     unstash 'CmdStanSetup'
-    writeFile(file: "make/local", text: "CXX = ${CXX}\n")
+    errorStr = failOnError ? "-Werror " : ""
+    writeFile(file: "make/local", text: "CXX=${CXX} ${errorStr}")
 }
 
-def runTests(String prefix = "") {
+def runTests(String prefix = "", String testPath) {
```
Comment (seantalts) [inline Jenkinsfile]: we don't need to pass the path in here :P

File: Jenkinsfile [hunk 2]
```diff
@@ -3,17 +3,23 @@ import org.stan.Utils
 
 def utils = new org.stan.Utils()
 
-def setupCXX(CXX = env.CXX) {
+def setupCXX(failOnError = true, CXX = env.CXX) {
     unstash 'CmdStanSetup'
-    writeFile(file: "make/local", text: "CXX = ${CXX}\n")
+    errorStr = failOnError ? "-Werror " : ""
```
Comment (seantalts) [inline Jenkinsfile]: we don't need this

**Comment by seantalts** [comment]
Thanks!

Build failed, maybe it needed the newline? Oh, I think you can't have default arguments that look in the environment, because it doesn't get populated when the script is first imported. If that's right you'll need to set the default to something like null and then check if it's null and set it to env.CXX or whatever.

**Comment by serban-nicusor-toptal** [comment]
I think default arguments that look into `env.` should work just fine.
It's like that in Stan Jenkinsfile for the Windows part.

**Comment by seantalts** [comment]
Where do you see that?
To clarify I mean this part:
https://github.com/stan-dev/cmdstan/pull/706/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R6

**Comment by serban-nicusor-toptal** [comment]
My bad, it was actually in cmdstan:
https://github.com/stan-dev/cmdstan/blob/develop/Jenkinsfile#L6

**Comment by seantalts** [comment]
Ah okay, maybe that does work now (though I swear I had problems with it a few years ago). Is it possible the Linux EC2 nodes don't have GCC env var defined?

**Comment by seantalts** [comment]
oops, no it failed on a Mac. hmm.

**Comment by serban-nicusor-toptal** [comment]
Hmh, the Macs do not have the `GCC` var set.

**Comment by seantalts** [comment]
Oh right, we want to run on Clang (e.g. CXX) on Macs and GCC on Linux only.

File: Jenkinsfile [hunk 3]
```diff
@@ -48,30 +53,50 @@ pipeline {
...
+                    }
                 }
-                stage('Non-windows interface tests') {
-                    agent any
+
+                stage('Linux interface tests') {
+                    agent {label 'linux'}
                     steps {
-                        setupCXX()
+                        setupCXX("${env.GCC}")
```
Comment (seantalts) [inline Jenkinsfile]: don't need to string interpolate this, you can just pass env.GCC here

**Comment by serban-nicusor-toptal** [comment]
Ah! Give me one sec to move Macs to CXX

File: Jenkinsfile [hunk 4]
```diff
@@ -48,30 +53,50 @@ pipeline {
...
-                    post { always { deleteDir() }}
+                    post { 
+                        always { 
+
+                            recordIssues id: "Windows",
+                            name: "Windows interface tests",
+                            enabledForFailure: true,
+                            aggregatingResults : true,
+                            tools: [
+                                gcc4(id: "Windows_gcc4", name: "Windows interface tests@GCC4"),
```
Comment (seantalts) [inline Jenkinsfile]: This `id` thing is very cool, did not know how to do that before and we got kind of messy results from it. Thanks :D

**Comment by seantalts** [comment]
Need to replace "&" with "&&" in label AND operation

**Pull request merged**